server/repository/src/db_diesel/json_schema.rs

```diff
@@ -0,0 +1,101 @@
+use super::StorageConnection;
+
+use crate::RepositoryError;
+
+use diesel::prelude::*;
+use serde::{Deserialize, Serialize};
+
+table! {
+    json_schema (id) {
+        id -> Text,
+        schema -> Text,
+    }
+}
+
+#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq)]
+#[table_name = "json_schema"]
+pub struct JSONSchemaRow {
+    /// The json schema id
+    pub id: String,
+    /// Document path and name
+    pub schema: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct JSONSchema {
+    pub id: String,
+    pub schema: serde_json::Value,
+}
+
+pub struct JsonSchemaRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+fn schema_from_row(schema_row: JSONSchemaRow) -> Result<JSONSchema, RepositoryError> {
+    let parsed_schema: serde_json::Value =
+        serde_json::from_str(&schema_row.schema).map_err(|err| RepositoryError::DBError {
```

Comment:
as per comment about common serialise/deserialise method, this could be inline

Reply:
ok I see what you mean now, I could also write a From impl...

Reply:
I guess if it's common enough we can use From right ? Although it would still be a bit of indirection, and you wouldn't be able to specify 'msg'. Probably ok as is, i was being too picky